pkg/controller/hostpathprovisioner/storagepool.go

```diff
@@ -408,6 +409,7 @@ func (r *ReconcileHostPathProvisioner) storagePoolDeploymentByNode(logger logr.L
 							},
 							SecurityContext: &corev1.SecurityContext{
 								Privileged: &privileged,
+								RunAsUser:  pointer.Int64Ptr(0),
```

Comment:
Have we considered how this affects PSA?

Reply:
It doesn't, we have to run as root since we are mounting volumes on the host. The problem is we were not running as root and ergo the mount failed. So we would use the host filesystem instead of the volume. This was actually introduced by trying to tighten permissions for PSA too much.

Reply:
Done